repair/row_level.cc

```diff
@@ -491,9 +491,16 @@ using is_local_reader = bool_class<class is_local_reader_tag>;
     future<mutation_fragment_opt>
     read_mutation_fragment() {
         ++_reads_issued;
-        return _reader(db::no_timeout).then([this] (mutation_fragment_opt mfopt) {
+        // Use a very long timeout for the reader to break out any eventual
+        // deadlock within the reader. Thirty minutes should be more than
+        // enough to read a single mutation fragment.
+        return _reader(db::timeout_clock::now() + std::chrono::minutes(30)).then([this] (mutation_fragment_opt mfopt) {
             ++_reads_finished;
             return mfopt;
+        }).handle_exception([this] (std::exception_ptr ep) {
```

Comment:
Please merge the exception handling into then_wrapped so we don't add an extra continuation.

Also we should drop the continuation that just does ++_reads_finished, but that's a preexisting problem.

Really, repair can't expect to be fast if it adds continuations eveywhere.

Reply:
Also, catch the expected type of exception. Otherwise the error message will be confusing.

Reply:
Continuations are everywhere in scylla too. And handle_exception is the standard way to handle the exception. It is sad we need to handle some usages differently. 

Anyway, since this is on the fast path, I will switch to then_wrappped.